auction-house/program/src/auctioneer/delegate.rs

```diff
@@ -51,21 +51,27 @@ pub fn delegate_auctioneer<'info>(
     }
 
     let auction_house = &mut ctx.accounts.auction_house;
-    auction_house.has_auctioneer = true;
-    auction_house.auctioneer_pda_bump = *ctx
-        .bumps
-        .get("ah_auctioneer_pda")
-        .ok_or(AuctionHouseError::BumpSeedNotInHashMap)?;
 
-    let auctioneer = &mut ctx.accounts.ah_auctioneer_pda;
-    auctioneer.auctioneer_authority = ctx.accounts.auctioneer_authority.key();
-    auctioneer.auction_house = ctx.accounts.auction_house.key();
+    if auction_house.has_auctioneer {
+        return Err(AuctionHouseError::AuctionHouseAlreadyDelegated.into());
+    }
+
+    auction_house.has_auctioneer = true;
+    auction_house.auctioneer_address = ctx.accounts.ah_auctioneer_pda.key();
 
     // Set all scopes false and then update as true the ones passed into the handler.
-    auctioneer.scopes = [false; MAX_NUM_SCOPES];
+    auction_house.scopes = [false; MAX_NUM_SCOPES];
```

Comment:
Nit: could you add an explanation for the change in scopes or tell me about it offline?

Reply:
We need to check the scopes in all the auctioneer instructions, but I don't think we're not passing in the auctioneer PDA everywhere so retrieving it from here is necessary for backwards compatibility.